### PR TITLE
test(ts-resolver): add unit tests for enrichment logic

### LINKED ISSUE #1284: follow-up: add unit tests for ts-resolver enrichment logic
Deferred from PR #1278 review.

**Original reviewer comment:** Claude (round 3 review) noted that `ts-resolver.ts` lacks unit test coverage for key logic paths:
- Two-pass disambiguation (same identifier, different types → ambiguous → skip)
- Lazy-load caching sentinel (`undefined` = not tried, `null` = unavailable)
- Solution-style tsconfig detection (empty `fileNames` → null + debug warning)
- `isTsFile` .d.ts exclusion

**Context:** The module was added in PR #1278 (Phase 8.1 TypeScript native type resolution). Test coverage was deferred to keep the PR focused on the feature itself. A minimal test with a mock tsconfig + simple .ts fixture would prevent regressions in the enrichment logic.

### PR DESCRIPTION
**Pull request by carlos-alm**
## Summary

- Adds 12 new tests across three describe blocks for `enrichTypeMapWithTsc`
- **Phase 8.1 typeMap enrichment** (6 tests — previously zero coverage): typed parameters → typeMap, typed variables → typeMap, confidence-1.0 guard, low-confidence replacement, ambiguous-name exclusion, primitive filtering via `SKIP_TYPE_NAMES`
- **callAssignment receiver type & deduplication** (3 tests): `receiverTypeName` captured from typeMap for method calls, pre-seeded varName skipped, `enrichSourceFile`→`enrichCallAssignments` deduplication integration
- **Edge cases** (3 tests): `.tsx` files processed, regular function expressions in `returnTypeMap`, primitive return types excluded

Closes #1284

## Test plan

- [x] All 21 tests pass (`npx vitest run tests/unit/ts-resolver.test.ts`)
- [x] No new lint warnings introduced

### CODE REVIEW

File: tests/unit/ts-resolver.test.ts [hunk 1]
```diff
@@ -259,3 +259,285 @@ class MyService {
...
+    );
+
+    const fileSymbols = makeFileSymbols(srcFile);
+    await enrichTypeMapWithTsc(tmpDir, fileSymbols);
+
+    const symbols = fileSymbols.get(srcFile)!;
+    const ca = symbols.callAssignments!.find((c) => c.varName === 'result');
+    expect(ca).toBeDefined();
+    expect(ca!.calleeName).toBe('getData');
+    expect(ca!.receiverTypeName).toBe('DataService');
```
[the remark this answers is not in the document]
Comment (carlos-alm) [reply tests/unit/ts-resolver.test.ts:423]: Fixed: added `expect(symbols.callAssignments).toBeDefined()` before each `!` assertion at lines 419, 441, and 461 (now shifted by a few lines). This ensures a clean vitest failure message if tsconfig setup ever drifts, rather than an opaque TypeError.

File: tests/unit/ts-resolver.test.ts [hunk 2]
```diff
@@ -259,3 +259,285 @@ class MyService {
...
+describe('enrichTypeMapWithTsc edge cases', () => {
+  let tmpDir: string;
+
+  beforeEach(() => {
+    tmpDir = makeTmpDir();
+  });
+
+  afterEach(() => {
+    fs.rmSync(tmpDir, { recursive: true, force: true });
+  });
```
[the remark this answers is not in the document]
Comment (carlos-alm) [reply tests/unit/ts-resolver.test.ts:484]: Fixed: added a default tsconfig.json write to the edge-cases beforeEach (matching the Phase 8.1 and callAssignment blocks). The .tsx test now explicitly overrides it with a comment explaining why. The two .ts tests that wrote the same config inline have had those redundant writes removed since beforeEach now covers them.

**Comment by carlos-alm** [comment]
@greptileai
